bn_mp_to_radix.c

```diff
@@ -16,14 +16,11 @@ mp_err mp_to_radix(const mp_int *a, char *str, size_t maxlen, size_t *written, i
    mp_digit d;
    char   *_s = str;
 
-
-   /* If we want to fill a bucket we need a bucket in the first place. */
-   if (str == NULL) {
-      return MP_VAL;
-   }
```

Comment:
This check was wrong. It is a programming error, not an error we should handle. I had the same discussion with fperrad a while ago, where he suggested checking every mp_int* for NULL. It does not make sense since we have no means of detecting valid/invalid memory addresses, and NULL is just an invalid address, as is probably 42. This is different than checking maxlen!

Reply:
I'd really love to see the architecture where NULL is valid memory, but I get the point.